src/pages/account/story/StoryEdit.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect } from "react";
+import { Redirect, useParams } from "react-router-dom";
+
+import axios from "axios";
+import axiosApi from "../../../utils/AxiosApi";
+import FieldSet from "../../../components/Form/Fieldset";
+import Input from "../../../components/Form/Input";
+import TextArea from "../../../components/Form/TextArea";
+import Button from "../../../components/Button";
+import NavBar from "../../../components/NavBar";
+
+const StoryEdit = () => {
+  const [input, setInput] = useState({});
+  const [story, setStory] = useState();
+  const [loading, setLoading] = useState(false);
+  const [redirect, setRedirect] = useState(false);
+  const [mediaPreview, setMediaPreview] = useState("");
+
+  const { storyId } = useParams();
+
+  useEffect(() => {
+    const fetchStoryData = async () => {
+      const result = await axiosApi.get(`/storyDetails/${storyId}`);
+      const singleStory = await result.data;
+      setStory(singleStory);
+    };
+    fetchStoryData();
+  }, [storyId]);
+
+  const handleImageUpload = async () => {
+    const data = new FormData();
+    data.append("file", input.media);
+    data.append("upload_preset", "idnmaxun");
+    const response = await axios.post(
+      process.env.REACT_APP_CLOUDINARY_URL,
+      data
+    );
+    const mediaUrl = await response.data.url;
+    input.media = mediaUrl;
+  };
+
+  const handleChange = (event) => {
+    const { name, value, files } = event.target;
+    if (name === "media") {
+      setInput((prevState) => ({ ...prevState, media: files[0] }));
+      setMediaPreview(window.URL.createObjectURL(files[0]));
+    } else {
+      setInput((prevState) => ({ ...prevState, [name]: value }));
+    }
+  };
+
+  const handleOnSubmit = async (event) => {
+    event.preventDefault();
+    try {
+      setLoading(true);
+      input.media && (await handleImageUpload());
```

Comment:
Like I said in my previous comment, return the url so you can use it here.
So it will look like something like this:

```javascript
try {
 setLoading(true);
 const imageUrl = await handleImageUpload();
 const response = await axiosApi.post("editStory", {
        storyData: {
        ...input, // spread it into a new object, because you need to enhance the object with the img url
        imageUrl
        },
        storyId,
   });
  setRedirect(response.status === 200)
}
```

Also notice that you should also check if the response went okay. Now you're always setting the redirect to `true`, but should that always be the case? What if something went wrong?
If correct, you're already sending a response back from the back-end